### PR TITLE
LocalProcessor supports filtering in Derived/SlidingFeatureView

### PR DESCRIPTION
**Pull request by Sxnan**
## What is the purpose of the change

Update LocalProcessor to support filtering in DerivedFeatureView and SlidingFeatureView.

## Brief change log

- Update LocalProcessor to support filtering in DerivedFeatureView and SlidingFeatureView
- Fix the bug in the LocalProcessor ast evaluator to handle IS NULL for numpy floating and python native float type

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by Sxnan** [comment]
@yunfengzhou-hub Could you review this PR?

File: python/feathub/processors/local/ast_evaluator/tests/test_local_ast_evaluator.py [hunk 1]
```diff
@@ -112,6 +114,9 @@ def test_is_op(self):
         self.assertEqual(False, self._eval("a IS NULL", {"a": 1}))
         self.assertEqual(False, self._eval("a IS NOT NULL", {"a": None}))
         self.assertEqual(True, self._eval("a IS NOT NULL", {"a": 1}))
+        self.assertEqual(False, self._eval("a IS NOT NULL", {"a": np.NAN}))
```
Comment (yunfengzhou-hub) [inline python/feathub/processors/local/ast_evaluator/tests/test_local_ast_evaluator.py:117]: Could you please also add an e2e test in `DerivedFeatureViewITTest` that evaluates `NAN` is/is not `NULL`? I hope to be guaranteed that judging `NAN` as `NULL` is widely accepted in Flink and Spark as well.

File: python/feathub/processors/local/ast_evaluator/local_ast_evaluator.py [hunk 2]
```diff
@@ -155,7 +155,10 @@ def eval_group_node(self, ast: GroupNode, variables: Optional[Dict]) -> Any:
 
     def eval_is_op(self, ast: IsOp, variables: Optional[Dict]) -> Any:
         left_value = self.eval(ast.left_child, variables)
-        is_none = left_value is None or np.isnan(left_value)
+        is_none = left_value is None
+        if isinstance(left_value, float) or isinstance(left_value, np.floating):
```
Comment (yunfengzhou-hub) [inline python/feathub/processors/local/ast_evaluator/local_ast_evaluator.py:159]: Do we need to judge from `np.int8` to `np.float64`?

File: python/feathub/processors/local/local_processor.py [hunk 3]
```diff
@@ -611,3 +612,7 @@ def _get_dependent_features(self, feature_view: FeatureView) -> Sequence[Feature
             if feature not in dependent_features:
                 dependent_features.append(feature)
         return dependent_features
+
+    def _filter_df(self, df: pd.DataFrame, filter_expr: str) -> pd.DataFrame:
```
Comment (yunfengzhou-hub) [inline python/feathub/processors/local/local_processor.py:616]: Would it be better to rename this method to `_filter_dataframe`? I noticed that none of the other methods use abbreviations in its signature.

**Review by yunfengzhou-hub — COMMENTED** [review]
Thanks for the PR. Left some comments as below.

File: python/feathub/processors/local/ast_evaluator/local_ast_evaluator.py (hunk 2 above)
Comment (Sxnan) [reply python/feathub/processors/local/ast_evaluator/local_ast_evaluator.py:159]: Thanks for the suggestion. We can check if the `left_value` is an instance of `np.generic`. This takes care of all the numpy dtypes.

File: python/feathub/processors/local/ast_evaluator/tests/test_local_ast_evaluator.py (hunk 1 above)
Comment (Sxnan) [reply python/feathub/processors/local/ast_evaluator/tests/test_local_ast_evaluator.py:117]: I think Flink and Spark don't have the notion of NAN. I added the test case where a float-type column has a null value.

**Comment by Sxnan** [comment]
@yunfengzhou-hub Thanks for your comments! I updated the PR. Could you take another look?

File: python/feathub/processors/local/ast_evaluator/tests/test_local_ast_evaluator.py (hunk 1 above)
Comment (yunfengzhou-hub) [reply python/feathub/processors/local/ast_evaluator/tests/test_local_ast_evaluator.py:117]: I noticed that Spark SQL has the notion of NAN[1], which is a numeric value and not equal to null. It might be better to follow this practice in LocalProcessor. Maybe we can add some function named `ISNAN()` to judge the corresponding situation in future.

[1] https://spark.apache.org/docs/3.3.1/sql-ref-datatypes.html#nan-semantics

**Comment by yunfengzhou-hub** [comment]
Thanks for the update. LGTM. @lindong28 could you please take another look?

**Comment by Sxnan** [comment]
@lindong28 I updated the PR according to our offline discussion. Could you please take another look?

**Comment by lindong28** [comment]
Thanks for the PR. LGTM.
